plugin/azure/README.md

```diff
@@ -0,0 +1,83 @@
+# azure
+
+## Name
+
+*azure* - enables serving zone data from Microsoft Azure DNS service.
+
+## Description
+
+The azure plugin is useful for serving zones from resource record
+sets in Microsoft Azure DNS. The `azure` plugin can be used when coredns is deployed on Azure or elsewhere.
+
+## Syntax
+
+~~~ txt
+azure [resource-group:dns-zone] {
```

Comment:
(i'm going over this PR slowly - hence a trickly of comments)

This syntax need to be styled different, see route53 readme for instance:
`<azure_tenant_id>` -> `TENTANT_ID` (for instance). 
Also not sure if prefixing everything with `azure` helps in readability.